ansible/roles/jenkins-node/tasks/main.yaml

```diff
@@ -0,0 +1,60 @@
+- name: Create jenkins group.
+  group:
+    name: jenkins
+
+- name: Create jenkins user.
+  user:
+    createhome: yes
+    group: jenkins
+    home: /home/jenkins
+    name: jenkins
+
+- name: Make .ssh dir for jenkins
+  file:
+    path: /home/jenkins/.ssh
+    state: directory
+    mode: 0700
+    owner: jenkins
+    group: jenkins
+
+- name: Install ssh key for jenkins
+  authorized_key:
+    key: "{{ lookup('file', '{{ jenkins_public_key }}' ) }}"
+    state: present
+    user: jenkins
+
+- name: Allow jenkins to run sudo commands without password.
+  lineinfile:
+    path: /etc/sudoers
+    state: present
+    regexp: '^%jenkins\s'
+    line: '%jenkins ALL=(ALL) NOPASSWD: ALL'
+
+- name: Install custom repo file
```

Comment:
I think you can make this optional by checking if the custom_repo_file is set so it can be copied. Also I think that the lookup could be used just like it is used to find the public ssh key to add to the authorized keys.